sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -620,6 +620,30 @@ object PushFoldableIntoBranches extends Rule[LogicalPlan] with PredicateHelper {
   }
 }
 
+/**
+ * Remove duplicated case when branches.
+ */
+object RemoveDuplicatedBranches extends Rule[LogicalPlan] with PredicateHelper {
+
+  private def contains(branches: Seq[(Expression, Expression)], elem: (Expression, Expression)) = {
+    branches.exists { case (condExpr, valueExpr) =>
+      condExpr.semanticEquals(elem._1) && valueExpr.semanticEquals(elem._2)
+    }
```

Comment:
Could we support another case? ```CASE WHEN cond1 THEN values1 WHEN cond1 THEN values2 END``` => ```CASE WHEN cond1 THEN values1 END```.

Reply:
Sure, but we must verify the behaviours on other DB.